Bottom: cb80f2110d69f2d7694e5947dd4ab35fbb0c42d3
Top:    ab2e21ba2373e686d04136b2a4c080cabad24069
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-11 01:02:19 +0100

Refresh of attempt-to-debug-islandness-af

---

diff --git a/src/action.c b/src/action.c
index 1886ef0..400025d 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1507,12 +1507,19 @@ NotifyMode (void)
 		   */
 		  SaveUndoSerialNumber ();
 		  Flags = ((PolygonType *)Crosshair.AttachedObject.Ptr2)->Flags;
-		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
-				      result, Flags);
+#warning FIXME: Removing a pour should force a re-check of islanding!
+                  /* For now, do this before PolyToPours() which re-inits the
+                   * clipping for the pour(s) affected (?). Ideally RemoveObject
+                   * should fix any island response its self. Anyway.. removing
+                   * the old thing first saves us doing twice the work by
+                   * recomputing islands when it gets deleted.
+                   */
 		  RemoveObject (POUR_TYPE,
 				Crosshair.AttachedObject.Ptr1,
 				Crosshair.AttachedObject.Ptr2,
 				Crosshair.AttachedObject.Ptr3);
+		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
+				      result, Flags);
 		  RestoreUndoSerialNumber ();
 		  IncrementUndoSerialNumber ();
 		  Draw ();
diff --git a/src/pour.c b/src/pour.c
index 2100377..d4590d0 100644
--- a/src/pour.c
+++ b/src/pour.c
@@ -1371,12 +1371,8 @@ InitPourClip (DataTypePtr Data, LayerTypePtr layer, PourType * pour)
     }
   while ((pg = tmp) != start_pg);
 
-  printf ("Checking islandiness\n");
   POURPOLYGON_LOOP (pour);
   {
-    printf ("Checking polygon from pour, %s\n",
-            IsPolygonAnIsland (layer, polygon) ? "is an island" :
-                                                 "connected");
     ASSIGN_FLAG (HOLEFLAG, IsPolygonAnIsland (layer, polygon), polygon);
   }
   END_LOOP;
